src/Workspaces/CSharp/Portable/CodeGeneration/MethodGenerator.cs

```diff
@@ -94,23 +112,47 @@ internal static class MethodGenerator
-                expressionBody: default,
-                semicolonToken: hasNoBody ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : new SyntaxToken());
+                    attributeLists: GenerateAttributes(method, options, explicitInterfaceSpecifier != null),
+                    modifiers: GenerateModifiers(method, destination, workspace, options, extractLocalFunction: false),
+                    returnType: method.GenerateReturnTypeSyntax(),
+                    explicitInterfaceSpecifier: explicitInterfaceSpecifier,
+                    identifier: method.Name.ToIdentifierToken(),
+                    typeParameterList: GenerateTypeParameterList(method, options),
+                    parameterList: ParameterGenerator.GenerateParameterList(method.Parameters, explicitInterfaceSpecifier != null, options),
+                    constraintClauses: GenerateConstraintClauses(method),
+                    body: hasNoBody ? null : StatementGenerator.GenerateBlock(method),
+                    expressionBody: default,
+                    semicolonToken: hasNoBody ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : new SyntaxToken());
 
             methodDeclaration = UseExpressionBodyIfDesired(workspace, methodDeclaration, parseOptions);
-
             return AddFormatterAndCodeGeneratorAnnotationsTo(methodDeclaration);
         }
 
+        private static LocalFunctionStatementSyntax GenerateLocalMethodDeclarationWorker(
+            IMethodSymbol method, CodeGenerationDestination destination,
+            Workspace workspace, CodeGenerationOptions options, ParseOptions parseOptions)
+        {
+            // Don't rely on destination to decide if method body should be generated.
+            // Users of this service need to express their intention explicitly, either by  
+            // setting `CodeGenerationOptions.GenerateMethodBodies` to true, or making 
+            // `method` abstract. This would provide more flexibility.
+            var hasNoBody = !options.GenerateMethodBodies || method.IsAbstract;
+
+            var localMethodDeclaration = SyntaxFactory.LocalFunctionStatement(
+                    modifiers: GenerateModifiers(method, destination, workspace, options, extractLocalFunction: true),
+                    returnType: method.GenerateReturnTypeSyntax(),
+                    identifier: method.Name.ToIdentifierToken(),
+                    typeParameterList: GenerateTypeParameterList(method, options),
+                    parameterList: ParameterGenerator.GenerateParameterList(method.Parameters, isExplicit: false, options),
+                    constraintClauses: GenerateConstraintClauses(method),
+                    body: hasNoBody ? null : StatementGenerator.GenerateBlock(method),
+                    expressionBody: default,
+                    semicolonToken: hasNoBody ? SyntaxFactory.Token(SyntaxKind.SemicolonToken) : new SyntaxToken());
```

Comment:
dedent.  indentation should never be more than '4' spaces (unless you're aligning with something on the previous line).